test/commands/provar/automation/metadata/download.nut.ts

```diff
@@ -1,194 +1,332 @@
+import {
+  errorMessages,
+  commandConstants,
+  SfProvarCommandResult,
+  setNestedProperty,
+} from '@provartesting/provardx-plugins-utils';
+import { Global } from '@salesforce/core';
 import * as validateConstants from '../../../../assertion/validateConstants.js';
 import * as metadataDownloadConstants from '../../../../assertion/metadataDownloadConstants.js';
 
 describe('sf provar config metadataDownload NUTs', () => {
-  let session: TestSession;
   const DOWNLOAD_ERROR = 'Error (1): [DOWNLOAD_ERROR]';
-  enum FILE_PATHS {
-    METADATA_ERROR_FILE = 'metadataErrorFile.json',
-    MALFORMED_METADATA_FILE = 'malformedMetadataFile.json',
-    METADATA_DOWNLOAD_FILE = 'metadataDownloadFile.json',
-  }
-
-  after(async () => {
-    await session?.clean();
-    Object.values(FILE_PATHS).forEach((filePath) => {
-      fileSystem.unlink(filePath, (err) => {
-        if (err) {
-          return err;
-        }
-      });
-    });
+  it('Update Provar Home and ProjectPath in provar dx properties file', async () => {
+    const ProvarDXPropertiesFilePath1 = path.join(process.cwd(), './provardx-properties.json');
+
+    // Read the config.json file
+    const ProvarDXPropertiesFileData = await fileSystem.readFile(ProvarDXPropertiesFilePath1, { encoding: 'utf8' });
+
+    // Parse the JSON data
+    /* eslint-disable */
+    const ProvarDXPropertiesFileParsed = JSON.parse(ProvarDXPropertiesFileData);
+    ProvarDXPropertiesFileParsed['provarHome'] = path.join(process.cwd(), './ProvarHome').replace(/\\/g, '/');
+    ProvarDXPropertiesFileParsed['projectPath'] = path
+      .join(process.cwd(), './ProvarRegression/AutomationRevamp')
+      .replace(/\\/g, '/');
+
+    setNestedProperty(ProvarDXPropertiesFileParsed, 'metadata.metadataLevel', 'Reuse');
+
+    // Convert the updated JSON object back to a string
+    const updatedPropertiesFileData = JSON.stringify(ProvarDXPropertiesFileParsed, null, 4);
+
+    // Write the updated JSON string back to the config.json file
+    await fileSystem.writeFile(ProvarDXPropertiesFilePath1, updatedPropertiesFileData, 'utf8');
   });
 
-  it('Missing file error as json file is not loaded', () => {
-    execCmd<SfProvarCommandResult>(
-      `${commandConstants.SF_PROVAR_AUTOMATION_CONFIG_GENERATE_COMMAND} -p ${FILE_PATHS.METADATA_ERROR_FILE}`
-    );
-    const jsonFilePath = FILE_PATHS.METADATA_ERROR_FILE;
-    const data = fileSystem.readFileSync(jsonFilePath, 'utf-8');
-    const newData = data.substring(1);
-    fileSystem.writeFile(jsonFilePath, newData, (error) => {
-      if (error) {
-        return;
+  it('It should update the PROVARDX_PROPERTIES_FILE_PATH in sf config.json', async () => {
+    // Call the function to perform the update
+    await UpdateFileConfigSfdx();
+    async function UpdateFileConfigSfdx(): Promise<void> {
+      try {
+        // Read the directory and find the config.json file
```

Comment:
I think we can remove all these comments at all places as the code is self-describing.

Reply:
Removed comment from all files